source/extensions/filters/http/oauth2/oauth_client.h

```diff
@@ -44,8 +44,9 @@ class OAuth2Client : public Http::AsyncClient::Callbacks {
 class OAuth2ClientImpl : public OAuth2Client, Logger::Loggable<Logger::Id::oauth2> {
 public:
   OAuth2ClientImpl(Upstream::ClusterManager& cm, const envoy::config::core::v3::HttpUri& uri,
+                   const envoy::config::route::v3::RetryPolicy& retry_policy,
                    const std::chrono::seconds default_expires_in)
-      : cm_(cm), uri_(uri), default_expires_in_(default_expires_in) {}
+      : cm_(cm), uri_(uri), retry_policy_(retry_policy), default_expires_in_(default_expires_in) {}
```

Comment:
Needn't copy the retry_policy here. Because the FilterConfig will have longer lifetime than the client anyway. Keep a copy of `OptRef<const RouteRetryPolicy>` is enough.

---
The OAuth2ClientImpl introduce lots of unnecessary data copy. But it's no problem of this PR. orz.